geoextract/__init__.py

```diff
@@ -434,10 +437,9 @@ def _augment_result(self, result):
         '''
         # Denormalize names
         for key in 'name', 'street', 'city':
-            try:
-                result[key] = self.normalized_names[result[key]]['name']
-            except KeyError:
-                pass
+            if key not in result or result[key] not in self.normalized_names:
+                continue
+            result[key] = self.normalized_names[result[key]]['name']
```

Comment:
In Python, exceptions are used differently than in many other languages (see, for example, [this post](https://softwareengineering.stackexchange.com/a/351121/233918)). In particular, Python follows the ["ask for forgiveness, not permission" paradigm](https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html). This means that catching exceptions is preferred to a priori checks. Hence, this change and the other similar ones are not an improvement.